### ISSUE TITLE
[java] LawOfDemeter: False positive with indexed array access

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2181]
https://github.com/pmd/pmd/issues/2160#issuecomment-565639733
**Affects PMD Version:** 
6.20.0

**Rule:**
[LawOfDemeter](https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter)

**Description:**
Index of array is treated as a method chain call by this rule. Data structures must be exempt from this rule especially indexed array access.

**Code Sample demonstrating the issue:**

```java
public final class Util {  
  
  public static boolean check(String passwd,
                              String hashed) {
    try {
      final String[] parts = hashed.split("\\$");

      if (parts.length != 5 || !parts[1].equals("s0")) {
        throw new IllegalArgumentException(
            "Invalid hashed value");
      }
    }
  }

  
}

```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
